aea/context/base.py

```diff
@@ -94,12 +92,12 @@ def addresses(self) -> Dict[str, str]:
     @property
     def address(self) -> str:
         """Get the defualt address."""
-        return self._addresses[FETCHAI] if FETCHAI in self._addresses.keys() else self._addresses[DEFAULT]
+        return self._addresses[self.ledger_apis.default_ledger_id]
```

Comment:
this is where this default_ledger bites. It sets the default addresses and public keys.